server/graphql/v2/object/HostMetricsTimeSeries.ts

```diff
@@ -58,13 +71,69 @@ export const HostMetricsTimeSeries = new GraphQLObjectType({
       resolve: async ({ host, account, dateFrom, dateTo, timeUnit }) => {
         let collectiveIds;
         if (account) {
-          const collectives = await fetchAccountsWithReferences(account);
+          const collectives = await fetchAccountsWithReferences(account, { attributes: ['id'] });
           collectiveIds = collectives.map(collective => collective.id);
         }
         const timeSeriesParams = { startDate: dateFrom, endDate: dateTo, collectiveIds, timeUnit };
         const results = await HostMetricsLib.getTotalMoneyManagedTimeSeries(host, timeSeriesParams);
         return { dateFrom, dateTo, timeUnit, nodes: resultsToAmountNode(results) };
       },
     },
+    totalReceived: {
+      type: new GraphQLNonNull(TimeSeriesAmountWithKind),
+      description: 'History of the total money received by this host',
+      resolve: async ({ host, account, dateFrom, dateTo, timeUnit }) => {
+        let collectiveIds;
+        if (account) {
+          const collectives = await fetchAccountsWithReferences(account, { attributes: ['id'] });
+          collectiveIds = collectives.map(collective => collective.id);
+        }
+
+        const amountDataPoints = await HostMetricsLib.getTransactionsTimeSeries(
+          [TransactionKind.CONTRIBUTION, TransactionKind.ADDED_FUNDS],
+          TransactionTypes.CREDIT,
+          host.id,
+          timeUnit,
+          collectiveIds,
+          dateFrom,
+          dateTo,
+        );
+
+        return {
+          dateFrom: dateFrom || host.createdAt,
+          dateTo: dateTo || new Date(),
+          timeUnit,
+          nodes: resultsToAmountWithKindNode(amountDataPoints),
+        };
+      },
+    },
+    totalSpent: {
+      type: new GraphQLNonNull(TimeSeriesAmountWithKind),
+      description: 'History of the total money spent by this host',
+      resolve: async ({ host, account, dateFrom, dateTo, timeUnit }) => {
+        let collectiveIds;
+        if (account) {
+          const collectives = await fetchAccountsWithReferences(account, { attributes: ['id'] });
+          collectiveIds = collectives.map(collective => collective.id);
+        }
+
+        const amountDataPoints = await HostMetricsLib.getTransactionsTimeSeries(
+          TransactionKind.EXPENSE,
+          TransactionTypes.DEBIT,
```

Comment:
Except for these two lines, everything else is the same within the `resolve` function in `totalReceived` (starting line 86). Maybe we can put them into a common function to avoid the duplication? 🤔

Reply:
It's a copy-paste of the resolvers introduced with `expenseAmountOverTime` + `contributionAmountOverTime`, which had pretty good factorization already (`fetchAccountsWithReferences`, `resultsToAmountWithKindNode`, `getTransactionsTimeSeries` are great helpers). This code takes a lot of lines, but mainly because of formatting.

I however agree that some parts could be factorized more, so I've added a small [change](https://github.com/opencollective/opencollective-api/pull/7433/commits/36f5375c7f8ed9adaa9f44dce16714134c43ecfe):
- Move `args.account` resolution up in `hostMetricsTimeSeries` resolver (since we need the same for all timeseries anyway)
- Stop trying to generate defaults for `dateFrom`/`dateTo`, the parent's again taking care of it
- Move all optional args for `getTotalMoneyManagedTimeSeries` to an object